### ISSUE TITLE
[Question] Low TPU Utilization for tranformer_big_enfr_tpu

### ISSUE THREAD

**Issue by Boffee** [tensorflow/tensor2tensor#1205]
### Description

I tried training a transformer model for `translate_enfr_wmt32k_packed` on a single TPU-v2 (8 cores)  using the default `transformer-transformer_big_enfr_tpu` hparams and noticed the TPU is heavily underutilized. The `cloud_tpu_profiler` tool reported only 18% TPU utilization because 69% of the time was spent on `OutfeedEnqueueTuple` operations. Looking through the trace_viewer, it seems that `OutfeedEnqueueTuple` is some sort of transition from loss to gradient computation because it is always preceded by a cross entropy op and followed by a softmax gradient op.

How can I reduce the time spent in `OutfeedEnqueueTuple` and what is it actually doing that takes so long? This is the only documentation I was able to find on it: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/tpu/ops/outfeed_ops.cc#L67.

### Environment information

```
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.5 LTS
Release:	16.04
Codename:	xenial
Base Image:	tensorflow/tensorflow:1.11.0-py3
GKE Version:	1.11.2-gke.9
Machine Type:	n1-standard-8
TPU Version:	v2

$ pip freeze | grep tensor
tensor2tensor==1.9.0
tensorboard==1.11.0
tensorflow==1.11.0

$ python -V
Python 2.7.12

$ python3 -V
Python 3.5.2
```

### Steps to reproduce:
```
t2t-trainer \
  --model=transformer \
  --problem=translate_enfr_wmt32k_packed \
  --hparams_set=transformer_big_enfr_tpu \
  --data_dir=$DATA_DIR \
  --output_dir=$OUT_DIR \
  --use_tpu=True \
  --cloud_tpu_name=$KUBE_GOOGLE_CLOUD_TPU_ENDPOINTS

capture_tpu_profile \
  --service_addr $KUBE_GOOGLE_CLOUD_TPU_ENDPOINTS \
  --logdir $OUT_DIR \
  --duration_ms 60000
```

### Logs:
**Input pipeline analysis:**
![image](https://user-images.githubusercontent.com/8808565/48108584-90aa0280-e1f8-11e8-8715-86232522592a.png)

**TPU Utilization:**
![image](https://user-images.githubusercontent.com/8808565/48108489-2abd7b00-e1f8-11e8-86b6-56f7666caaf5.png)

**Full trace:**
![image](https://user-images.githubusercontent.com/8808565/48108541-60626400-e1f8-11e8-8218-e73ea78216d5.png)

**Op before OutfeedEnqueue:**
![image](https://user-images.githubusercontent.com/8808565/48108562-740dca80-e1f8-11e8-85ec-a7d8d54e1f0a.png)

**Op after OutfeedEnqueue:**
![image](https://user-images.githubusercontent.com/8808565/48108555-6b1cf900-e1f8-11e8-9cd2-650fccfd503e.png)

**Comment by Boffee** [comment]
Looks like this was caused by the `host_call` function in the `TPUEstimator`, which was always enabled in tensor2tensor version 1.9.0 and now defaults to disabled in version 1.10.0.

From my basic understanding, the outfeed queue is used to store the output of XLA compiled graphs that they can be access by other ops and XLA graphs. The `host_call` function is slow because it copies the data from the TPU to the host machine (CPU) on very iteration, which is very expensive.
